agent/functional_tests/tests/functionaltests_test.go

```diff
@@ -635,6 +635,58 @@ func testV3TaskEndpoint(t *testing.T, taskName, containerName, networkMode, awsl
 	assert.Equal(t, 42, exitCode, fmt.Sprintf("Expected exit code of 42; got %d", exitCode))
 }
 
+// testContainerMetadataFile validates that the metadata file from the
+// ECS_CONTAINER_METADATA_FILE environment variable contains all the required
+// fields
+func testContainerMetadataFile(t *testing.T, taskName, awslogsPrefix string) {
+	agentOptions := &AgentOptions{
+		ExtraEnvironment: map[string]string{
+			"ECS_ENABLE_CONTAINER_METADATA": "true",
+			"ECS_AVAILABLE_LOGGING_DRIVERS": `["awslogs"]`,
+		},
+	}
+
+	agent := RunAgent(t, agentOptions)
+	defer agent.Cleanup()
+	agent.RequireVersion(">=1.15.0")
```

Comment:
since this test also checks for HostIPv4 field, shouldn't the agent version required be the most recent one?

Reply:
Good catch. I think we should move away from requiring a version and represent these "requirements" as capabilities. For functional tests, all tests should run against the most up to date version of Agent.